src/server.rs

```diff
@@ -397,8 +397,14 @@ impl<O: Output> LsService<O> {
         self.output.success(id, ResponseData::Init(result));
 
         let root_path = params.root_path.map(PathBuf::from).expect("No root path");
+        let init_options: Option<InitializationOptions> = params.initialization_options
+                              .and_then(|options| serde_json::from_value(options)
```

Comment:
Is deserialisation more likely to fail here than elsewhere? I would prefer the simpler version I think - this is kinda hard to read.